### PR TITLE
fix: Replace shortid package with nanoid package

### PR DESCRIPTION
**Pull request by AkshatJawne**
Resolves https://github.com/deephaven/web-client-ui/issues/1922

Changes Implemented:
- Replaced shortid with nanoid, given that shortid is deprecated (Reran all tests to ensure that functionality upholds)

Info on Nanoid: https://github.com/ai/nanoid/blob/main/README.md

### CODE REVIEW

**Comment by AkshatJawne** [comment]
Looking into errors here

**Comment by mattrunyon** [comment]
Remove `esmify` from golden-layout dev deps and add `babelify` as a dev dependency.

In `karma.conf.cjs` remove the `plugins` line and add this. For some reason `esmify` isn't transforming node_modules even though it says it does (it might not understand the package.json of nanoid).

```js
transform: [
  ['babelify', { presets: ['@babel/preset-env'], global: true }],
],
```

This seems to fix the issue by using babel to transpile the ESM before the bundling for the test.

**Comment by bmingles** [comment]
LGTM. Will wait for @mattrunyon to approve since he suggested the babelify change, and I'm less familiar here

**Comment by AkshatJawne** [comment]
Sounds good, thanks @bmingles

File: packages/golden-layout/karma.conf.cjs [hunk 1]
```diff
@@ -21,10 +21,13 @@ module.exports = function (config) {
     preprocessors: {
       // '../src/**': 'coverage'
       './test/helper.js': ['browserify'],
+      nanoid: ['browserify'],
```
Comment (mattrunyon) [inline packages/golden-layout/karma.conf.cjs:24]: This shouldn't be needed

File: packages/golden-layout/package.json [hunk 2]
```diff
@@ -50,8 +50,8 @@
     "access": "public"
   },
   "devDependencies": {
+    "babelify": "^10.0.0",
```
Comment (mattrunyon) [inline packages/golden-layout/package.json:53]: golden-layout should have `nanoid` in its normal dependencies. Looks like `shortid` wasn't added there initially which was a mistake

File: packages/golden-layout/package.json [hunk 3]
```diff
@@ -53,6 +53,7 @@
     "babelify": "^10.0.0",
     "browserify": "^17.0.0",
     "karma-browserify": "^8.1.0",
+    "nanoid": "^5.0.7",
```
[the remark this answers is not in the document]
Comment (AkshatJawne) [reply packages/golden-layout/package.json:56]: Ahh makes sense, sorry about that. Pushing changes now

**Review by mattrunyon — COMMENTED** [review]
There's a `@types/shortid` in the root `package.json` still. Remove it and run `npm install` again so it's removed from package-lock too.

**Review by mattrunyon — APPROVED** [review]

**Pull request merged**